js/src/components/SettingsReplacementVariableEditors.js

```diff
@@ -22,13 +24,28 @@ class SettingsReplacementVariableEditors extends React.Component {
 		super( props );
 	}
 
+	filterEditorSpecificReplaceVars( replacementVariables, pageType ) {
+		const {
+			editor_specific_replace_vars: editorSpecificReplaceVars = {},
+			shared_replace_vars: sharedReplaceVars,
+		} = wpseoReplaceVarsL10n;
+
+		const pageTypeSpecificReplaceVars = editorSpecificReplaceVars[ pageType ] || [];
+		const replaceVarNames = [ ...sharedReplaceVars, ...pageTypeSpecificReplaceVars ];
+
+		return replacementVariables.filter( replaceVar => {
+			return replaceVarNames.includes( replaceVar.name );
+		} );
+	}
+
 	/**
 	 * Renders all replacement variable editor components.
 	 *
 	 * Renders a settings replacement variable editor in every given element in
 	 * the editorElements props. It requires every element to have a data-react-
-	 * replacevar-title-field-id, data-react-replacevar-metadesc-field-id
-	 * attribute and data-react-replacevar-page-type to function properly.
+	 * replacevar-title-field-id, data-react-replacevar-metadesc-field-id,
+	 * data-react-replacevar-page-type-recommend and
+	 * data-react-replacevar-page-type-specific attributes to function properly.
 	 * These attributes should point to existing (hidden) inputs in the DOM.
```

Comment:
I guess this line should now be changed to clarify _which_ attributes should point to hidden fields.